docs/guide/interactions/checks.rst

```diff
@@ -0,0 +1,368 @@
+
+.. code-block:: python3
+    :emphasize-lines: 6,17
+
+    # Custom check function.
+    def check_if_owner(interaction: discord.Interaction) -> bool:
+        return interaction.guild is not None and interaction.user.id == interaction.guild.owner_id
+
+    @app_commands.command()
+    @app_commands.check(check_if_owner)
+    async def owners_only(interaction: discord.Interaction):
+        await interaction.response.send_message('Hello boss!')
+
+    # Custom check decorator.
+    def is_me():
+        def predicate(interaction: discord.Interaction) -> bool:
+            return interaction.user.id == 1234567890
+        return app_commands.check(predicate)
+
+    @app_commands.command()
+    @is_me()
+    async def me_only(interaction: discord.Interaction):
+        await interaction.response.send_message('I know you!', ephemeral=True)
+
+In this example, the first check is implemented to only allow the command usage if the user who runs the command is the owner of the guild the command is used in. This is simply a function which takes a :class:`.Interaction` as an argument, which is passed to a :func:`@check <.app_commands.check>` decorator. The second check implemented creates a custom decorator which checks if the user who activates a command has a specific id. This limits a command to only one user.
+
+.. _guide_interaction_checks_combining-checks:
+
+Combining Checks
+-----------------
+
+You can combine multiple checks on a command to end up with a very particular functionality for your needs, in a very concise manner.
+
+As an example, here is a command which only runs if a user has a certain role, ``send_messages`` permissions, the bot has ``manage_messages`` permissions, and with a ``15`` second global cooldown:
+
+.. code-block:: python3
+
+    @app_commands.command()
+    @app_commands.checks.has_role('Necessary Role')
+    @app_commands.checks.has_permissions(send_messages=True)
+    @app_commands.checks.bot_has_permissions(manage_messages=True)
+    @app_commands.checks.cooldown(1, 15.0, key=None)
+    async def check_example(interaction: discord.Interaction):
+        await interaction.send_message('Checks passed!', ephemeral=True)
+
+.. _guide_interaction_checks_global-checks:
+
+Global Interaction Checks
+--------------------------
+
+In addition to adding individual or combined checks onto specific app commands or context menus, it is also possible to create global interaction checks. For these, any interaction
+passed through the bot for either app commands or context menus will first run this check.
+
+In order to implement a global check, we will need to create our own subclass of the :class:`.app_commands.CommandTree` class, and implement the :meth:`.app_commands.CommandTree.interaction_check` method.
+
+.. code-block:: python3
+
+    import discord
+    from discord.ext import commands
+
+    class MyCommandTree(discord.app_commands.CommandTree):
```

Comment:
I went back and forth on this one a few times...I was worried that having a collection of imports might distract from the intention of showing the rest of the code. But the way it is currently is quite long and can be confusing as well, so i wasnt sure. probably better to go with your route tbh

Reply:
I'm not sure i agree with this here. Consistency is better than conciseness, and it can get confusing to throw names around in code snippets. I could see importing app_commands instead, but i wouldn't go further than that

Reply:
Definitely agree with Tom. The [examples](https://github.com/Rapptz/discord.py/tree/master/examples/app_commands) utilize an app_commands import. Feels like it'd be better to use that for consistency.

Reply:
I like this middle ground, and keeping in line with the examples makes sense. I will update when I get home from work.

Reply:
Added this one in!